scripts/y2c

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+#
+# Author: Cyrus Durgin <cyrusd@gmail.com>
+#
+
+trap "exit 1;" SIGINT SIGTERM
+
+read -r -d "" HELP << "EOF"
+usage: y2c [-i INPUT_DIRECTORY] [-o OUTPUT_DIRECTORY] [-p PATH]
+
+Generates Python classes from YANG models.
+
+optional arguments:
+  -i INPUT_DIR          Read YANG models from this directory root.
+  -o OUTPUT_DIR         Write Python classes to this directory root.
+  -p PATH               Append PATH to the pyang -p option.
+EOF
+
+INPUT_DIR="models"
+OUTPUT_DIR="classes"
+
+PYANG=`which pyang`
+if [ -z "$PYANG" ]; then
+    echo "Cannot locate pyang; install with pip." >&2
+    exit 1
+fi
+
+PYBINDPLUGIN=`/usr/bin/env python -c 'import pyangbind; import os; print "%s/plugin" % os.path.dirname(pyangbind.__file__)'`
+if [ -z "$PYBINDPLUGIN" ]; then
+    echo "Cannot locate pyangbind; install with pip." >&2
+    exit 1
+fi
+
+while getopts ":hi:o:p:" opt; do
+    case $opt in
+        h)
+            echo "$HELP" >&2
+            exit 0
+            ;;
+        i)
+            INPUT_DIR=$OPTARG
+            ;;
+        o)
+            OUTPUT_DIR=$OPTARG
+            ;;
+        p)
+            PYANG_PATH=$OPTARG
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            echo >&2
+            echo "$HELP" >&2
+            exit 1
+            ;;
+        :)
+            echo "Option -$OPTARG requires an argument." >&2
+            echo >&2
+            echo "$HELP" >&2
+            exit 1
+            ;;
+    esac
+done
+
+YANG_SRC_FILES=$(find $INPUT_DIR -type f -a -name "*.yang" 2> /dev/null)
+
+if [ $? -gt 0 ]; then
+    echo "Cannot find yang models in '$INPUT_DIR'." >&2
+    echo >&2
+    echo "$HELP" >&2
+    exit 1
+fi
+
+ERRORS=0
+SRC_FILES=0
+SECONDS=0
+ERROR_FILES=()
+
+for YANG_SRC_FILE in $YANG_SRC_FILES; do
+    ((SRC_FILES++))
+
+    CLASS_DIR=$(echo $(dirname $(realpath $YANG_SRC_FILE)) | sed "s#$PWD/$INPUT_DIR##" | sed "s#^/##")
```

Comment:
Should `CLASS_DIR` replace the base path - and instead mirror the structure after the INPUT_DIR - so, for example, if I specify `-i /Users/rjs/Code/openconfig/public/releases` and `-o /tmp/oc` - it'd be good to get `vlans/openconfig-vlan.yang` to output to `/tmp/oc/vlan` rather than to `/tmp/oc/Users/rjs/Code/...`.